tonic-web/src/layer.rs

```diff
@@ -24,7 +24,7 @@ impl Default for GrpcWebLayer {
 
 impl<S> Layer<S> for GrpcWebLayer
 where
-    S: Service<http::Request<hyper::Body>, Response = http::Response<BoxBody>>,
+    S: Service<http::Request<BoxBody>, Response = http::Response<BoxBody>>,
```

Comment:
I think the alternative is to use `hyper::body::Incoming` here, but that makes testing and re-using `tonic-web` much harder, because `Incoming` cannot be constructed outside of `hyper`.

Reply:
probably just makes sense to not rely on incoming tbh

Reply:
So should we have our own `Incoming` type maybe? Not necessarily a fan of the type erasure approach of just using `BoxBody` all over the place.

Reply:
I think the Body story is pretty fragmented. Internally, we are converting to and from `axum::body::Body` when using the `transport`. Axum just wraps a `BoxBody` in their own tuple struct. Effectively so does reqwest. We could add our own body to the mix, and we might get some benefits for enum-dispatch over one layer, but the next layer is gonna hit the v-table.